File: Server_Root/Server/Server/JoinLobbyToC.cpp
#include "stdafx.h"
#include "JoinLobbyToC.h"
#include "NetworkDefines.h"
#include "JSONHelper.h"


JoinLobbyToC::JoinLobbyToC()
{
	m_buffer = new GenericStringBuffer<UTF8<>>();
	m_writer = new Writer<StringBuffer, UTF8<>>(*m_buffer);
}

JoinLobbyToC::~JoinLobbyToC()
{
	delete m_buffer;
	delete m_writer;
}

unsigned short JoinLobbyToC::GetID()
{
	return (unsigned short)Messages::Join_Lobby_ToC;
}

const char* JoinLobbyToC::Serialize()
{
	Document document;
	document.SetObject();

	JSONHelper::AddField(&document, "Result", m_nResult);

	document.Accept(*m_writer);

	return m_buffer->GetString();
}

bool JoinLobbyToC::Deserialize(const char* pChar)
{
	Document document;
	document.Parse<0>(pChar);
	if (!document.IsObject())
	{
		return false;
	}

	if (!JSONHelper::GetField(&document, "Result", &m_nResult)) return false;

	return true;
}

File: Client_Root/Client/Assets/Scripts/Network/Messages/Lobby/ToClient/JoinLobbyToC.cs
using System.Text;

public class JoinLobbyToC : IMessage
{
    public const ushort MESSAGE_ID = 10000;

    public int m_nResult;     //  json field name : Result

    public ushort GetID()
    {
        return MESSAGE_ID;
    }

    public byte[] Serialize()
    {
        JSONObject jsonObj = new JSONObject(JSONObject.Type.OBJECT);

        JSONHelper.AddField(jsonObj, "Result", m_nResult);

        return Encoding.Default.GetBytes(jsonObj.Print());
    }

    public bool Deserialize(byte[] bytes)
    {
        JSONObject jsonObj = new JSONObject(Encoding.UTF8.GetString(bytes));

        if(!JSONHelper.GetField(jsonObj, "Result", ref m_nResult)) return false;

        return true;
    }
}
